pkg/component/controller/cplb/cplb_linux.go

```diff
@@ -398,7 +404,11 @@ func (k *Keepalived) watchReconcilerUpdatesReverseProxy(ctx context.Context) err
 func (k *Keepalived) setProxyRoutes() {
 	routes := []tcpproxy.Route{}
 	for _, addr := range k.reconciler.GetIPs() {
-		routes = append(routes, tcpproxy.To(fmt.Sprintf("%s:%d", addr, k.APIPort)))
+		if k.PrimaryAFIPv4 {
+			routes = append(routes, tcpproxy.To(fmt.Sprintf("%s:%d", addr, k.APIPort)))
+		} else {
+			routes = append(routes, tcpproxy.To(fmt.Sprintf("[%s]:%d", addr, k.APIPort)))
+		}
```

Comment:
This might be better handled via `net.JoinHostPort`

Reply:
Not only it's cleaner, this also solves a bug I hadn't considered in dual stack clusters.